src/provider/CoinbaseWalletProvider.ts

```diff
@@ -347,7 +351,9 @@ export class CoinbaseWalletProvider
   }
 
   public close() {
-    void this.initializeRelay().then(relay => relay.resetAndReload());
+    void this.initializeRelay().then(relay =>
+      this._headlessMode ? relay.reset() : relay.resetAndReload()
```

Comment:
```suggestion
      if (this._headlessMode) {
        relay.reset();
      } else {
        relay.resetAndReload();
      }
```
value selection operators shouldn't be abused for flow control :-)